### PR TITLE
cleanup clippy tests

### PR DESCRIPTION
**Pull request by svenski123**
#### Problem
Clippy complaints in test code have clept in again.

#### Summary of Changes
Clean up the clippy complaints

### CODE REVIEW

**Comment by mvines** [comment]
Strange, `test_vote_subscribe` seems to be failing 2 out of 2 times with this PR 🤔

**Comment by ryoqun** [comment]
@mvines Well, I thought `rpc_pubsub::tests::test_vote_subscribe` has been flaky for some time.

**Comment by ryoqun** [comment]
Thanks for working on this again!!

@svenski123 How about doing this in this pr?: https://github.com/solana-labs/solana/pull/10030#issuecomment-629246364:

> > `cargo check` and `cargo clippy --tests` are clean.
> 
> @svenski123 Wow. What a stamina you have! Legend thanks!! So, are you confident and ready to receive my dearest honor of adding `--tests` here?:
> 
> https://github.com/solana-labs/solana/blob/6a6c5f196a71ffcdf342846a847da6efd18997e4/ci/test-checks.sh#L30


Also, [your concern](https://github.com/solana-labs/solana/issues/5503#issuecomment-629392530) at #5503  is fixed already. :)

from [my comment there](https://github.com/solana-labs/solana/issues/5503#issuecomment-630493699):

> We rather want to enable --tests (or equivalent) sooner or later otherwise we'll introduce more clippy errors because CI won't complain, including me ;)

I guessed this will occur. xD

Also, you don't need to run all of ci tests locally, just believe your edit and test it on our ci. For these tiny edits this is enough.

File: core/src/rpc_pubsub.rs [hunk 1]
```diff
@@ -895,8 +895,7 @@ mod tests {
         let (subscriber, _id_receiver, receiver) = Subscriber::new_test("voteNotification");
 
         // Setup Subscriptions
-        let subscriptions =
-            RpcSubscriptions::new(&exit, bank_forks.clone(), block_commitment_cache.clone());
+        let subscriptions = RpcSubscriptions::new(&exit, bank_forks, block_commitment_cache);
```
Comment (ryoqun) [inline core/src/rpc_pubsub.rs:898]: @svenski123 Could you check the flakiness of this test `test_vote_subscribe` locally? it seems like this test got a lot unstable with removing `.clone()` on ci:

- https://buildkite.com/solana-labs/solana/builds/24611#4e149382-e000-4a94-85c3-c8f520e42009/467-993
- https://buildkite.com/solana-labs/solana/builds/24611#b6c44f63-29ad-41a5-a030-c16c22e8dc06/128-654
- https://buildkite.com/solana-labs/solana/builds/24611#b16ec14f-7317-477a-a337-a09bacd3a411/133-659
- https://buildkite.com/solana-labs/solana/builds/24611#b38fbc2e-0e10-485c-9133-9109113aedc7/116-642

My guess is that this or [that](https://github.com/solana-labs/solana/pull/10172/files#diff-9078d14bfb5de677e818bdca3c32a292R928) `.clone()` takes some time and removing that makes it more susceptible to the some kind of race condition. If you're brave enough, feel free to go down the rabbit hole or just add `#allow(...)`...

FYI, this test has been a bit flaky for some time, iirc.

**Comment by svenski123** [comment]
Having removed clippy complaints before, I obviously couldn't reintroduce them again in #9980 (though not using a type alias for tuple-of-hash-of-integer-to-vector-of-struct wasn't my idea :)).

So I tugged on that thread and some more stuff fell out - hence this PR.

test_vote_subscribe fails locally - seems one of the channels gets closed prematurely, and this is somehow linked to the earlier drop due to one of the missing clones. So this testcase needs to be picked apart a bit, gone through and put together (rather than simply sticking the clone back in).

**Comment by ryoqun** [comment]
> Having removed clippy complaints before, I obviously couldn't reintroduce them again in #9980 (though not using a type alias for tuple-of-hash-of-integer-to-vector-of-struct wasn't my idea :)).

hehe ;) I was pretty sure you'll introduce one for it.

> So I tugged on that thread and some more stuff fell out - hence this PR.

I see

> test_vote_subscribe fails locally - seems one of the channels gets closed prematurely, and this is somehow linked to the earlier drop due to one of the missing clones. So this testcase needs to be picked apart a bit, gone through and put together (rather than simply sticking the clone back in).

Hehe, it seems that you're interested. Thanks!

**Comment by ryoqun** [comment]
@svenski123 How is this going? :) Still trying to fix the root cause? Or am I making you to wait for something? You can fix it at your pace, but rahter I want to merge the rest of clippy nice fixes you did and `ci/test-check.sh` change first. :)

**Comment by svenski123** [comment]
Hi @ryoqun ! Apologies I've not had much time for development this week so this has been dragging out. Also there's quite a lot going on in that testcase and rust's various async / Future related libraries take some time to work through. I'll try to get to it this evening but if not I'll put the clone back that makes it work with a clippy override to unblock this.

**Comment by ryoqun** [comment]
> Hi @ryoqun ! Apologies I've not had much time for development this week so this has been dragging out.

I see! Please take your time. There should be no rush. :)

> Also there's quite a lot going on in that testcase and rust's various async / Future related libraries take some time to work through. I'll try to get to it this evening but if not I'll put the clone back that makes it work with a clippy override to unblock this.

Oh I see! I'm also not that knowledgeable with that test, too... Thanks for looking! Also, feel free ask any questions if that's makes it easy for you.

For now, let's merge the rest! We always welcome your contribution. :)

**Pull request merged**